src/instructlab/sdg/llmblock.py

```diff
@@ -18,6 +18,7 @@
 logger = logging.getLogger(__name__)
 
 DEFAULT_MAX_NUM_TOKENS = 4096
+DEFAULT_KNOWLEDGE_UPSAMPLE_AMOUNT = 5000
```

Comment:
Do we really want to default this to 5000 here? In the typical upstream scenario, users have constrained hardware and there's no precomputed skills dataset used in phase 2 that we need to balance against. Do we know how much longer this will take in the typical upstream train cycle for those new users kicking the tires?

Reply:
I think for the typical upstream usage we may not want to upsample at all, so that would be a default value of `None` I guess. Is there a compelling reason to upsample when there is no precomputed skills training data we need to balance against?